core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/CompactEntityProvider.scala

```diff
@@ -381,7 +381,39 @@ class CompactEntityProvider(requestArguments: EntityRequestArguments,
                             entityName: String,
                             newName: String,
                             parentContext: RawlsRequestContext
-  ): Future[Int] = ???
+  ): Future[Int] = {
+    // Check if the newName is the same as the old name
+    if (newName == entityName) {
+      throw new RawlsExceptionWithErrorReport(
+        errorReport = ErrorReport(StatusCodes.BadRequest, s"New name $newName is the same as the old name $entityName")
+      )
+    }
```

Comment:
nit: here and in the `EntityNotFoundException` below, we _probably_ shouldn't echo the $newName and $entityName in the error message. This allows the user to inject arbitrary text into an error message, which can lead to XSS. This is so minor I hesitate to bring it up but it's a good best practice.